opentelemetry-api/src/opentelemetry/context/context.py

```diff
@@ -16,6 +16,23 @@
 from abc import ABC, abstractmethod
 
 
+class Context:
+    def __init__(
+        self, values: typing.Optional[typing.Dict[str, object]] = None
```

Comment:
Little suggestion to use regular `kwargs` instead of `values`.

Reply:
In fact, I think this class can be implemented simply like this:

```python
class Context(dict):
    def __setitem__(self, key, value):
        raise ValueError
```
Here is an example that uses this class:
```python
# example.py


class Context(dict):
    def __setitem__(self, key, value):
        raise ValueError


class Object:
    pass

the_context_dict = Context(a=9, b=8, object_=Object())

print(the_context_dict["a"])
print(the_context_dict["b"])
print(the_context_dict["object_"])

the_context_dict_copy = the_context_dict.copy()

print(the_context_dict_copy["a"])
print(the_context_dict_copy["b"])
print(the_context_dict_copy["object_"])

# Notice how X["object_"] have the same memory address for
# the_context_dict_copy and the copy below made with dict
print(dict((key, value) for key, value in the_context_dict.items()))

try:
    the_context_dict["a"] = 3
except ValueError:
    print("Unable to assign a new value")

try:
    the_context_dict["z"] = 3
except ValueError:
    print("Unable to create a new key")
```
So, instead of having a method named `snapshot`, just use the `copy` method present already in a `dict`.

Reply:
Great suggestion, implemented this and added a test to ensure `Context` is immutable.